File: samples/dark_cyber.cs
/*
 * In a land shrouded in mystery...
 */

public class Cyber
{
  public string Name { get; private set; }
  public int ConstructionYear { get; private set; }
  public int DestructionYear { get; private set; }
  public string[] Features { get; private set; }

  public DarkCyber(string name, int constructionYear, int destructionYear, string[] features)
  {
    Name = name;
    ConstructionYear = constructionYear;
    DestructionYear = destructionYear;
    Features = features;
  }

  public int Age()
  {
    return calcAge();
  }

  private int calcAge()
  {
    return DestructionYear - ConstructionYear;
  }
}

class Program
{
  static void Main(string[] args)
  {
    // ...there stood a Dark Cyber

    var cyber = new DarkCyber(
      "Shadowhollow",
      1023,
      1598,
      new string[] { "Tall Towers", "Enchanted Forest" }
    );

    System.Console.WriteLine($"The age of {cyber.Name} is {cyber.Age()} years.");
  }
}

File: samples/dark_cyber.dart
/*
 * In a land shrouded in mystery...
 */

class DarkCyber {
  String name;
  int constructionYear, destructionYear;
  List<String> features;

  DarkCyber({this.name, this.constructionYear, this.destructionYear, this.features});

  int get age => this.calcAge();

  int calcAge() => this.destructionYear - this.constructionYear;
}

void main() {
  // ...there stood a Dark Cyber
  final cyber = DarkCyber(
      name: 'Nightfall',
      constructionYear: 1023,
      destructionYear: 1598,
      features: ['Tall Towers', 'Enchanted Forest']);

  print('The age of ${cyber.name} is ${cyber.age} years.');
}
